File: aoj/volume100/ruby/A10013.rb
$<.each do |line|
  h, w = line.split.map(&:to_i)
  break if h == 0 && w == 0

  edge   = "#"*w + "\n"
  inside = "#" + ("."*(w-2)) + "#\n" 
  frame  = edge + inside*(h-2) + edge + "\n"
  print frame
end

File: aoj/volume100/haskell/A10013.hs
import Control.Applicative

main = do
  inputs <- lines <$> getContents
  mapM_ putStrLn
    $ fmap (\(h, w) -> toFrame h w)
    $ fmap tuplify2
    $ fmap (fmap (read :: String -> Int) . words)
    $ takeWhile (/="0 0")
    $ inputs

toFrame h w =
  frame
  where
    edge   = repStr w "#" ++ "\n"
    inside = "#" ++ repStr (w-2) "." ++ "#\n"
    frame  = edge ++ repStr (h-2) inside ++ edge

repStr n s =
  concat $ take n $ repeat s

tuplify2 [a, b] = (a, b)
